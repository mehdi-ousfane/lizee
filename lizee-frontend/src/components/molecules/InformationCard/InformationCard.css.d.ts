declare const styles: {
  readonly "informationCard": string;
};
export = styles;

