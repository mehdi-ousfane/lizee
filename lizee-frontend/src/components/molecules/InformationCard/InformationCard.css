.informationCard {
  border: 1rem solid white;
  border-radius: 5px;
}
