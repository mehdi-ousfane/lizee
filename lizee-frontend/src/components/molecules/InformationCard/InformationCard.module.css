.informationCard {
  display: flex;
  border: 1px solid white;
  font-family: 'Patron webfont', sans-serif;
  font-size: 0.9rem;
  margin-right: 1rem;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  text-align: center;
  border: 0.1rem solid white;
  background-color: hsla(0, 0%, 8%, 0.4);
  border-radius: 1rem;
  cursor: pointer;
}

.informationCard:hover {
  background-color: hsla(0, 0%, 8%, 0.688);
}
