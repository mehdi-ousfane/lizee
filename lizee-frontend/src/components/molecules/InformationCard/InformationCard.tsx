import { ReactNode, MouseEvent } from 'react';

import styles from './InformationCard.module.css';

interface Props {
  children?: ReactNode;
  onClick?: (event: MouseEvent<HTMLDivElement>) => void;
}

const InformationCard = ({ children, onClick }: Props) => {
  return (
    <div className={styles.informationCard} onClick={onClick}>
      {children}
    </div>
  );
};

export default InformationCard;
