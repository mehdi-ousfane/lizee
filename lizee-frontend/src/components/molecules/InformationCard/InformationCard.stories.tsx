import { Meta } from '@storybook/react/types-6-0';
import InformationCard from './InformationCard';

const containerStyle = {
  display: 'flex',
  background: 'black',
};

export default {
  title: 'molecules/InformationCard',
  component: InformationCard,
} as Meta;

export const Rounded = () => (
  <div style={containerStyle}>
    <InformationCard>
      <p>Coins arrondis</p>
    </InformationCard>
  </div>
);
