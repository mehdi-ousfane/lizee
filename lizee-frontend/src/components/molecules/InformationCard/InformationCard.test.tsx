import InformationCard from './InformationCard';
import { render, fireEvent } from '@testing-library/react';

describe('<InformationCard />', () => {
  test('should match snapshots', () => {
    const { container } = render(<InformationCard />);
    expect(container).toMatchSnapshot();
  });

  test('should call onClick prop', () => {
    const onClick = jest.fn();
    const { container } = render(<InformationCard onClick={onClick} />);

    fireEvent.click(container.querySelector('div') as HTMLElement);

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
