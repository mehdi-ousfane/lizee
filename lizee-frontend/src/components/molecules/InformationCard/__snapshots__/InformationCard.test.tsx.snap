// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InformationCard /> should match snapshots 1`] = `
<div>
  <div
    class="informationCard"
  />
</div>
`;
